### PR TITLE
fix: Workaround JSON.stringify failing on large arrays with the error…

### PR DESCRIPTION
**Pull request by paddlefish**
…geError: Invalid string length" when downloading profiling

Summary:
---------

Fixes https://github.com/react-native-community/cli/issues/2050

Test Plan:
----------

Follow steps in issue 2050

Checklist
----------

- [x] Documentation is up to date to reflect these changes.
- [x] Follows commit message convention described in [CONTRIBUTING.md](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md#commit-message-convention)

### CODE REVIEW

File: packages/cli-hermes/src/profileHermes/downloadProfile.ts [hunk 1]
```diff
@@ -123,9 +123,14 @@ export async function downloadProfile(
         file,
         '.cpuprofile',
       )}-converted.json`;
+
+      // Convert to JSON in chunks because JSON.stringify() will fail for large
+      // arrays with the error "RangeError: Invalid string length"
+      const out = events.map(JSON.stringify).join(",");
```
Comment (thymikee) [inline packages/cli-hermes/src/profileHermes/downloadProfile.ts:129]: can we keep the space of 4 as in the original code?

File: packages/cli-hermes/src/profileHermes/downloadProfile.ts [hunk 2]
```diff
@@ -123,9 +123,14 @@ export async function downloadProfile(
...
+
       fs.writeFileSync(
         transformedFilePath,
-        JSON.stringify(events, undefined, 4),
+        "["+out+"]",
```
Comment (thymikee) [inline packages/cli-hermes/src/profileHermes/downloadProfile.ts:133]: please run `yarn lint --fix` to fix formatting

**Comment by szymonrybczak** [comment]
Hey @paddlefish, thanks for creating PR! If you don't have time for adjusting PR with @thymikee's suggestions - could you please allow edits from maintainers so that I can finish this PR, and we can get this merged?

**Review by szymonrybczak — APPROVED** [review]
Thank you!

**Comment by szymonrybczak** [comment]
CI Green, @thymikee ready to merge.

**Pull request merged**